### DISCUSSION TITLE
Can you use Pages to host a game?

### DISCUSSION THREAD

**Discussion by aagb1884** [community/community#138771] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm making a website with with two text-based-adventure games. They're free to play, they're React JSX and CSS only. First time I hosted one on Vercel I had to pay for hosting as it exceeded free tier bandwidth. I'm assuming this isn't something I can use GitHub Pages to host, the terms imply that I can't but don't explicitly say so, so I wanted to check first.

**Comment by davevad93** [comment]
Hi @aagb1884 ,

I've seen some games built with `React` hosted on GitHub Pages, for example this one is built with `Next.js` [2048-in-react](https://github.com/mateuszsokola/2048-in-react), I've even seen some `Godot` and `Unity` games hosted on Pages, so I think it's perfectly possible that you can host your game there too. 

Here's a list of [games hosted on GitHub Pages](https://games-site.github.io/) if you want to take a look.

**Comment by loglot** [comment]
i've been hosting games on GitHub for a while, but the way I do it is by making static websites with html and vanilla javascript

**Reply by aagb1884** [reply]
Have you ever had any issues with server traffic?

**Reply by loglot** [reply]
not really, all of my games are client side only games though, so all of the source code is easily accessible on the user end

**Comment by sctech-tr** [comment]
you can use it to host anything, as long as it does not break github's terms of service and you have the necessary knowledge to do so. (as far as i know)

**Reply by aagb1884** [reply]
the terms imply that I can't but don't explicitly say so, so I wanted to check first.

**Reply by davevad93** [reply]
@LiteBrite82 , is it possible that hosting a game on GitHub Pages violates [GitHub's Terms of Service](https://docs.github.com/site-policy/github-terms/github-terms-of-service)?

**Reply by loglot** [reply]
> the terms imply that I can't but don't explicitly say so, so I wanted to check first.

where in the terms did you see that?

**Reply by aagb1884** [reply]
> > the terms imply that I can't but don't explicitly say so, so I wanted to check first.
> 
> where in the terms did you see that?

"Each Account comes with access to the [GitHub Pages static hosting service](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages). GitHub Pages is intended to host static web pages, but primarily as a showcase for personal and organizational projects.

GitHub Pages is not intended for or allowed to be used as a free web hosting service to run your online business, e-commerce site, or any other website that is primarily directed at either facilitating commercial transactions or providing commercial software as a service (SaaS). Some monetization efforts are permitted on Pages, such as donation buttons and crowdfunding links." 

So I took that to be vague on whether an unpaid game fell into personal projects or if it could be considered promotional material.

**Reply by sctech-tr** [reply, marked as answer]
it would probably fall into the "personal projects" category

**Reply by loglot** [reply]
> GitHub Pages is not intended for or allowed to be used as a free web hosting service to run your online business, e-commerce site, or any other website that is primarily directed at either facilitating commercial transactions or providing commercial software as a service (SaaS). Some monetization efforts are permitted on Pages, such as donation buttons and crowdfunding links."

yeah, running a free game is perfectly fine, this is just referring to if you are running a business where you gain money through using github pages as a free webhost

**Reply by LiteBrite82** [reply]
Nothing wrong with sharing a project and showcasing your skills

**Reply by LiteBrite82** [reply]
As long as content doesn't violate ToS

**Comment by LiteBrite82** [comment]
@aagb1884 
Feel free to post the Games in `Pages`; we love to see what you built and how you did it!
